include/RAJA/policy/sycl/kernel/ForICount.hpp

```diff
@@ -0,0 +1,425 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Header file for SYCL statement executors.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-22, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+
+#ifndef RAJA_policy_sycl_kernel_ForICount_HPP
+#define RAJA_policy_sycl_kernel_ForICount_HPP
+
+#include "RAJA/config.hpp"
+
+#include "RAJA/policy/sycl/kernel/internal.hpp"
+
+
+namespace RAJA
+{
+
+namespace internal
+{
+
+
+
+/*
+ * Executor for local work sharing loop inside SyclKernel.
+ * Mapping directly from local id to indices
+ * Assigns the loop iterate to offset ArgumentId
+ * Assigns the loop count to param ParamId
+ */
+template <typename Data,
+          camp::idx_t ArgumentId,
+          typename ParamId,
+          int ThreadDim,
+          typename... EnclosedStmts,
+          typename Types>
+struct SyclStatementExecutor<
+    Data,
+    statement::ForICount<ArgumentId, ParamId, RAJA::sycl_local_012_direct<ThreadDim>, EnclosedStmts...>,
+    Types>
+    : public SyclStatementExecutor<
+        Data,
+        statement::For<ArgumentId, RAJA::sycl_local_012_direct<ThreadDim>, EnclosedStmts...>, Types> {
+
+  using Base = SyclStatementExecutor<
+        Data,
+        statement::For<ArgumentId, RAJA::sycl_local_012_direct<ThreadDim>, EnclosedStmts...>,
+        Types>;
+
+  using typename Base::enclosed_stmts_t;
+  using typename Base::diff_t;
+
+  static
+  inline
+  RAJA_DEVICE
+  void exec(Data &data, cl::sycl::nd_item<3> item, bool thread_active)
+  {
+    diff_t len = segment_length<ArgumentId>(data);
+    auto i = item.get_local_id(ThreadDim);
```

Comment:
Should this be diff_t?
I worry about getting warnings about comparing signed and unsigned in `(i<len)`.